### DISCUSSION TITLE
Some questions about data preprocessing

### DISCUSSION THREAD

**Discussion by Tonywhitemin** [mljar/mljar-supervised#557] [General]
Hi @pplonski,
Wish you have a good day!
Since I found that there are some good functions in mljar-AutoML for data preprocessing , but I don't know how to use it...
Could you help?
1. LeaveOneOutEncoder: The default setting is "categorical_to_int", May I know in what case will LeaveOneOutEncoder work?
![Screenshot from 2022-08-05 10-45-21](https://user-images.githubusercontent.com/88414200/182990688-93db0aa2-6129-4810-a56c-ce2dfc3dd985.png)

2. Fill missing values with mean value:The default setting is "FILL_NA_MEDIAN". Can we specify "FILL_NA_MEAN" or others instead?
![Screenshot from 2022-08-05 10-49-22](https://user-images.githubusercontent.com/88414200/182991157-47e14939-13b6-4deb-a78e-d78c70f07c7a.png)

3. Scale method for "scale_log_and_normal" : The default setting is "scale_normal". Can we specify "scale_log_and_normal" instead? Or in what case will "scale_log_and_normal"  work?
![Screenshot from 2022-08-05 10-54-17](https://user-images.githubusercontent.com/88414200/182992030-24460c6a-1e78-477e-802f-71a4266e92a4.png)

Thanks for your time!

**Comment by pplonski** [comment]
1. LeaveOneOutEncoder is not used - it was causing target leakage and overfitting.
2. FILL_NA_MEAN is not used - there is no big difference in results. You can try to overwrite default settings in your fork.
3. `scale_log_and_normal` is used when the target is skewed (there is a skew test done on the target).

**Reply by Tonywhitemin** [reply]
Thank you @pplonski !
